common/src/main/java/gov/cms/ab2d/common/service/UserServiceImpl.java

```diff
@@ -34,12 +36,30 @@ public User getCurrentUser() {
 
     @Override
     public User getUserByUsername(String username) {
-        return userRepository.findByUsername(username);
+        User user = userRepository.findByUsername(username);
+        if (user == null) {
+            String userNotPresentMsg = "User is not present in our database";
+            log.error(userNotPresentMsg);
+            throw new ResourceNotFoundException(userNotPresentMsg);
+        } else {
+            return user;
+        }
     }
 
     @Override
-    public User createUser(UserDTO userDTO) {
+    public UserDTO createUser(UserDTO userDTO) {
         User user = mapping.getModelMapper().map(userDTO, User.class);
-        return userRepository.saveAndFlush(user);
+        User createdUser = userRepository.saveAndFlush(user);
+        return mapping.getModelMapper().map(createdUser, UserDTO.class);
+    }
+
+    @Override
+    public UserDTO updateUser(UserDTO userDTO) {
+        // When updating there needs to be verification that the user exists
+        User user = getUserByUsername(userDTO.getUsername());
+        userDTO.setId(user.getId());
+        User mappedUser = mapping.getModelMapper().map(userDTO, User.class);
+        User updatedUser = userRepository.saveAndFlush(mappedUser);
```

Comment:
Are you creating a new user entity here?
It seems to me that a brand new User entity is being created by the mapping library from userDTO and being saved in line 61 & 62. 
Since the user object was already fetched by hibernate in line 59 above, shouldn't the properties in the managed entity be getting updated from the DTO instead?

Reply:
No, it uses ID of the user it has found